platform/android/MapboxGLAndroidSDKTestApp/src/androidTest/java/com/mapbox/mapboxsdk/maps/CameraUpdateFactoryTest.kt

```diff
@@ -0,0 +1,167 @@
+package com.mapbox.mapboxsdk.maps
+
+import android.support.test.runner.AndroidJUnit4
+import com.mapbox.mapboxsdk.camera.CameraPosition
+import com.mapbox.mapboxsdk.camera.CameraUpdateFactory
+import com.mapbox.mapboxsdk.geometry.LatLng
+import com.mapbox.mapboxsdk.geometry.LatLngBounds
+import com.mapbox.mapboxsdk.testapp.activity.EspressoTest
+import org.junit.Assert.assertEquals
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.TimeoutException
+
+@RunWith(AndroidJUnit4::class)
+class CameraUpdateFactoryTest : EspressoTest() {
```

Comment:
Can the resulting zoom level of the tests differ based on the pixel ratio? Shouldn't this test use the device independent test activity?

Reply:
nice catch, yes definitly